### PR TITLE
Docs: Add information on Environment Variables configuration

### PR DESCRIPTION
**Pull request by LiraLemur**
Signed-off-by: Sara <sara@Sara.Sara>

### Description of the Change

In https://github.com/hyperledger/iroha/pull/705 we introduced the option of setting up configuration via env variables instead of config file. Here I add some docs for that.

### Benefits

Documented change

### Possible Drawbacks 

Might need some elaboration?

### CODE REVIEW

File: docs/source/configure/index.rst [hunk 1]
```diff
@@ -146,6 +148,54 @@ Environment-specific parameters
   ``"initial_peers" : [{"address":"127.0.0.1:10001", "public_key":
   "bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308929"}]``
 
+Alternative
```
Comment (MBoldyrev) [inline docs/source/configure/index.rst:151]: I would avoid marking this possibility as "alternative", which sounds like "secondary". The approach with environment is meant to provide completely the same functionality as the config file, so IMO they should be explained as equally adequate approaches. What about making two same-level sections, where one describes config file syntax, and the other is about env vars?

File: docs/source/configure/index.rst [hunk 2]
```diff
@@ -146,6 +148,54 @@ Environment-specific parameters
...
+===========
+
+Alternative to using a configuration file is setting up the configuration using environment variables.
+**Configuration file, if present, will have a priority in parameters that are set up in it.**
```
Comment (MBoldyrev) [inline docs/source/configure/index.rst:155]:
```suggestion
**The parameters specified in the configuration file, if present, will override the ones that are set up through environment.**
```

File: docs/source/configure/index.rst [hunk 3]
```diff
@@ -146,6 +148,54 @@ Environment-specific parameters
   ``"initial_peers" : [{"address":"127.0.0.1:10001", "public_key":
   "bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308929"}]``
 
+Alternative
+===========
+
+Alternative to using a configuration file is setting up the configuration using environment variables.
```
Comment (MBoldyrev) [inline docs/source/configure/index.rst:154]: strictly speaking, it is not an alternative, as both can be used at once, and your next sentence is just about that

File: docs/source/configure/index.rst [hunk 4]
```diff
@@ -146,6 +148,54 @@ Environment-specific parameters
...
+**Configuration file, if present, will have a priority in parameters that are set up in it.**
+
+Here is how it will look like in
```
Comment (MBoldyrev) [inline docs/source/configure/index.rst:157]: I would suggest to add a bit more formal description on how to set some parameter through env.
first, the lazy approach, let the user set the flag `--verbosity trace` and take the parameter name from the log.
second, the names are formed from the config structure: fixed path labels are taken as is, arrays are indexed starting from `0`, and user-provided dictionary keys are a bit trickier: the key and the value are set in separate vars. everything is uppercased and joined with `_`.

File: docs/source/configure/index.rst (hunk 1 above)
Comment (LiraLemur) [reply docs/source/configure/index.rst:151]: Ok, will do!

File: docs/source/configure/index.rst [hunk 5]
```diff
@@ -146,6 +148,54 @@ Environment-specific parameters
...
+Windows
+-------
+
+``setx IROHA_BLOCK_STORE_PATH=/tmp/block_store/
+setx IROHA_TORII_PORT=50051
+setx IROHA_INTERNAL_PORT=10001
+.....
+``
+
+Shell
```
Comment (lebdron) [inline docs/source/configure/index.rst:190]:
```suggestion
PowerShell
```

**Comment by LiraLemur** [comment]
Should be fixed now @MBoldyrev  @lebdron

File: docs/source/configure/index.rst [hunk 6]
```diff
@@ -181,21 +184,80 @@ bin/irohad --genesis_block example/genesis.block --overwrite_ledger --verbosity=
...
+``"initial_peers" : [{"address":"127.0.0.1:10001", "public_key": "bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308929"}, {"address":"127.0.0.1:10002", "public_key": "bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308920"}] ``
+
+In env variables:
+
+``IROHA_INITIAL_PEERS_0_ADDRESS=127.0.0.1:10001
+IROHA_INITIAL_PEERS_0_PUBLIC_KEY=bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308929
+IROHA_INITIAL_PEERS_1_ADDRESS=127.0.0.1:10002
+IROHA_INITIAL_PEERS_1_PUBLIC_KEY=bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308920``
+
+**User-provided user dictionaries**
```
Comment (MBoldyrev) [inline docs/source/configure/index.rst:235]:
```suggestion
**User-provided user dictionaries of users**
```

File: docs/source/configure/index.rst [hunk 7]
```diff
@@ -181,21 +184,80 @@ bin/irohad --genesis_block example/genesis.block --overwrite_ledger --verbosity=
...
+
+User-provided dictionary keys are a bit trickier: the key and the value are set in separate vars
+
+In config file:
+
+``"crypto": {"providers": {
```
Comment (MBoldyrev) [inline docs/source/configure/index.rst:241]: maybe an example with loggers would be easier to grasp:

```
  "log": {
    "level": "info",
    "children": {
      "KeysManager": {
        "level": "trace"
      },
      "Irohad": {
        "level": "error"
      }
    }
  }
```

**Review by MBoldyrev — APPROVED** [review]

File: docs/source/configure/index.rst (hunk 7 above)
Comment (lebdron) [reply docs/source/configure/index.rst:241]: There is an example below

Comment (MBoldyrev) [reply docs/source/configure/index.rst:241]: indeed

File: docs/source/configure/index.rst (hunk 6 above)
Comment (LiraLemur) [reply docs/source/configure/index.rst:235]: Doesn't sound very nice. Can you please elaborate so I could probably try to make it look a bit better?...

Comment (LiraLemur) [reply docs/source/configure/index.rst:235]: you're joking, right?)

Comment (LiraLemur) [reply docs/source/configure/index.rst]: Should be better now :D

Comment (LiraLemur) [reply docs/source/configure/index.rst]: Still waiting for @lebdron

Comment (lebdron) [reply docs/source/configure/index.rst]: Simple string values and arrays are also user-provided, so it looked like a joke. I think "user-provided" part can be removed, leaving just "dictionaries".

**Comment by lebdron** [comment]
Can you please fix your username and email in local git settings?

**Comment by LiraLemur** [comment]
@lebdron  I fixed the dictionaries part and also the sign-off message. 
Is it ok now?

File: docs/source/configure/index.rst [hunk 8]
```diff
@@ -232,7 +232,7 @@ IROHA_INITIAL_PEERS_0_PUBLIC_KEY=bddd58404d1315e0eb27902c5d7c8eb0602c16238f00577
 IROHA_INITIAL_PEERS_1_ADDRESS=127.0.0.1:10002
 IROHA_INITIAL_PEERS_1_PUBLIC_KEY=bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308920``
 
-**User-provided dictionaries**
+**Dictionaries**
```
Comment (MBoldyrev) [inline docs/source/configure/index.rst:235]:
```suggestion
**Dictionaries with user-defined keys**
```

**Review by MBoldyrev — APPROVED** [review]

**Pull request merged**